Frameworks/CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -5007,21 +5007,35 @@ CFURLRef CFURLCreateCopyDeletingPathExtension(CFAllocatorRef allocator, CFURLRef
     }
 #endif
     
+    CFStringRef lastPathComponent = NULL;
     rg = _rangeOfLastPathComponent(url);
-    if (rg.location < 0) {
+    if ( (rg.location == kCFNotFound) || ((lastPathComponent = CFStringCreateWithSubstring(allocator, url->_string, rg)) == NULL) ) {
```

Comment:
>lastPathComponent = CFStringCreateWithSubstring [](start = 43, length = 47)

pretty please don't do assignments in conditionals. Especially not compound conditionals. Its real real confusing to figure out whats going on.